lib/prawn/embedded_files/embedded_file.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'digest/md5'
+
+module Prawn
+  module EmbeddedFiles
+    class EmbeddedFile
+      attr_reader :chksum
+
+      def initialize(data, options = {})
+        @creation_date = options[:creation_date]
+        unless @creation_date.instance_of?(Time)
+          @creation_date = Time.now.utc
+        end
+
+        @mod_date = options[:modification_date]
+        unless @mod_date.instance_of?(Time)
+          @mod_date = Time.now.utc
+        end
+
+        @chksum = Digest::MD5.hexdigest(data)
+        @data = data.dup
```

Comment:
Duplicating the data here seems wasteful - any particular reason? Especially, since it is just appended to the stream later on in line 36.

Reply:
Indeed, I was a little hesitant about duplicating or not. I was thinking about hypothetical cases that may never occur. Fixed